modules/internal/ChapelRange.chpl

```diff
@@ -2455,7 +2455,8 @@ operator :(r: range(?), type t: range(?)) {
       compilerError("iteration over a range with no first index");
 
     if followThis.size != 1 then
-      compilerError("iteration over a range with multi-dimensional iterator");
+      compilerError("rank mismatch in zippered iteration (leader is " +
+                    followThis.size:string + "D, but ranges are 1D)");
```

Comment:
This is surely an improvement. I suggest clarifying it further by using the word "follower", ex.
`(leader is 3D, but follower is a range, which is 1D)`

Reply:
(implied, but I should've said it:  So let me know whether you agree with the rationale for the current message, still prefer your original proposal, prefer something else, or would like me to create more of a design discussion around this).